### DISCUSSION TITLE
Cannot figure out how to apply permissions to commands

### DISCUSSION THREAD

**Discussion by Yorashoo** [discordjs/discord.js#7572] [Q&A] [answered]
<img width="731" alt="image" src="https://user-images.githubusercontent.com/100492642/155875204-d3c4ecc8-fbd3-4a80-8319-8a18fbe535cc.png">

This is the code from the discordjs.guide, but I just can't get this to work. I've spent about 4 hours messing around with it but I'm just not sure:

a) how to get a command's ID
b) how to apply "permissions" to a command
c) how to in general use this approach

I'm following the command structure created here (https://discordjs.guide/creating-your-bot/command-handling.html#reading-command-files).

Here's what my main.js looks like: https://pastebin.com/rz6BRs5R

Help would be greatly appreciated, thank you! (also, is there a discord or something I can join? I haven't been able to find one)

**Comment by BenjammingKirby** [comment, marked as answer]
> a) how to get a command's ID

```js
await guild.commands.fetch()
```
to fetch all the commands, afterwards you can use [`find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) to find the command via it's `name`
and now you have a singular specific command available to you 
now just accessing `command.id` would give you the id
to apply that entirely it would be
```js
const myKickCommandID = (await guild.commands.fetch()).find(command => command.name === "kick")?.id
```
> b) how to apply "permissions" to a command

after you've gotten a singular command, it's exactly like the guide, you call .permissions.add to it read the [`docs`](https://discord.js.org/#/docs/discord.js/stable/class/ApplicationCommandPermissionsManager?scrollTo=add) for more information on this
> c) how to in general use this approach

this requires you to already have commands registered and you run this code once to add the permissions, maybe on the ready event and then after setting the permissions you can remove it right after. 

> Help would be greatly appreciated, thank you! (also, is there a discord or something I can join? I haven't been able to find one)

Yes! Invite: https://discord.gg/djs
